torch/csrc/jit/export.cpp

```diff
@@ -540,14 +547,22 @@ void ScriptModuleSerializer::serialize(const script::Module& module) {
 
 void ScriptModuleSerializer::convertModel(
     const script::Module& module,
-    torch::ModelDef* model_def) {
+    torch::ModelDef* model_def,
+    const script::ExtraFilesMap& extra_files) {
   model_def->set_producer_name("pytorch");
   model_def->set_producer_version("1.0"); // TODO: set the producer version
                                           // using appropriate function call
   model_def->set_proto_version(torch::ProtoVersion::PROTO_VERSION_NEWEST);
   convertModule(
       module, "", writer_.archiveName(), model_def->mutable_main_module());
   writeTensorTable(model_def);
+
+  // Write out extra files.
+  for (const auto& kv : extra_files) {
+    const std::string key = "extra/" + kv.first;
+    model_def->add_extra_files()->set_key(key);
```

Comment:
I don't think we need to add an entry to torch.proto here. Avoiding it would allow other tools to just stash files in the extra folder without also having to edit the data in the model_def. We expect some users of meta-data will end up just adding files to the zip archive this way.